api/http/handler/auth/logout_oauth.go

```diff
@@ -0,0 +1,73 @@
+package auth
+
+import (
+	"errors"
+	. "github.com/portainer/libhttp/error"
+	"github.com/portainer/libhttp/response"
+	"io/ioutil"
+	"mime"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type verifyResponse struct {
+	Valid bool `json:"valid"`
+}
+
+func (handler *Handler) invalidateOAuthSession(w http.ResponseWriter, r *http.Request) *HandlerError {
+	defer r.Body.Close()
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return &HandlerError{http.StatusInternalServerError, "Unable to block token: ", err}
+	}
+
+	content, _, err := mime.ParseMediaType(r.Header.Get("Content-Type"))
+	if err != nil {
+		return &HandlerError{http.StatusInternalServerError, "Unable to find content-type ", err}
+	}
+
+	if content == "application/x-www-form-urlencoded" || content == "text/plain" {
+		values, err := url.ParseQuery(string(body))
+		if err != nil {
+			return &HandlerError{http.StatusInternalServerError, "Cannot parse body ", err}
+		}
+
+		token := values.Get("logoutRequest")
+
+		handler.JWTService.AddTokenToBlacklist(token)
+
+		w.WriteHeader(200)
+		return nil
+	} else {
+		return &HandlerError{http.StatusInternalServerError, "Invalid content type ", err}
```

Comment:
At this point `err` is most probably in an invalid state in terms of the current execution branch (most probably `nil`).

You could create your own error object. Also the used content type might be of interest to the person reading the error message.

Reply:
Done